CMakeLists.txt

```diff
@@ -13,14 +13,25 @@ FetchContent_Declare(
   GIT_REPOSITORY "https://github.com/ml-explore/mlx-c.git"
   GIT_TAG "v0.0.2")
 FetchContent_MakeAvailable(mlx-c)
+set(swift_numerics_patch git apply ${CMAKE_CURRENT_SOURCE_DIR}/cmake/swift-numerics.patch)
+FetchContent_Declare(
+  swift-numerics
+  GIT_REPOSITORY "https://github.com/apple/swift-numerics.git"
+  GIT_TAG "1.0.2"
+  PATCH_COMMAND ${swift_numerics_patch} || true
+)
+FetchContent_MakeAvailable(swift-numerics)
+
+# swift-numerics
 
 # MLX package
 file(GLOB MLX-src ${CMAKE_CURRENT_LIST_DIR}/Source/MLX/*.swift)
 add_library(MLX STATIC ${MLX-src})
 target_include_directories(MLX
                            PUBLIC ${CMAKE_CURRENT_LIST_DIR}/Source/Cmlx/include)
 # Note: mlxc is target name exported by mlx-c
-target_link_libraries(MLX PUBLIC mlxc)
+target_link_libraries(MLX PUBLIC mlxc Numerics)
+target_compile_options(MLX PRIVATE -package-name MLX)
```

Comment:
I wonder if this is needed for all the packages?

Reply:
It's needed only in places where you use
```swift
public package(set)
```

for now it looks like it's only in `Source/MLX/MLXArray.swift`